actors/builtin/miner/miner_test.go

```diff
@@ -3605,6 +3605,154 @@ func TestConfirmUpdateWorkerKey(t *testing.T) {
 	})
 }
 
+func TestChangeOwnerAddress(t *testing.T) {
+	actor := newHarness(t, 0)
+	builder := builderForHarness(actor).
+		WithBalance(bigBalance, big.Zero())
+	newAddr := tutil.NewIDAddr(t, 1001)
+	otherAddr := tutil.NewIDAddr(t, 1002)
+
+	t.Run("successful change", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.SetCaller(actor.owner, builtin.MultisigActorCodeID)
+		actor.changeOwnerAddress(rt, newAddr)
+
+		info := actor.getInfo(rt)
+		assert.Equal(t, actor.owner, info.Owner)
+		assert.Equal(t, newAddr, *info.PendingOwnerAddress)
+
+		rt.SetCaller(newAddr, builtin.MultisigActorCodeID)
+		actor.changeOwnerAddress(rt, newAddr)
+
+		info = actor.getInfo(rt)
+		assert.Equal(t, newAddr, info.Owner)
+		assert.Nil(t, info.PendingOwnerAddress)
+	})
+
+	t.Run("proposed must be valid", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		nominees := []addr.Address{
+			addr.Undef,
+			tutil.NewSECP256K1Addr(t, "asd"),
+			tutil.NewBLSAddr(t, 1234),
+			tutil.NewActorAddr(t, "asd"),
+		}
+		rt.SetCaller(actor.owner, builtin.MultisigActorCodeID)
+		for _, a := range nominees {
+			rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+				actor.changeOwnerAddress(rt, a)
+			})
+		}
+	})
+
+	t.Run("withdraw proposal", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.SetCaller(actor.owner, builtin.MultisigActorCodeID)
+		actor.changeOwnerAddress(rt, newAddr)
+
+		// Revert it
+		actor.changeOwnerAddress(rt, actor.owner)
+
+		info := actor.getInfo(rt)
+		assert.Equal(t, actor.owner, info.Owner)
+		assert.Nil(t, info.PendingOwnerAddress)
```

Comment:
An ExpectAbort when the proposed address attempts to confirm it would add confidence and clarity here.

Reply:
Done